openhands/core/config/security_config.py

```diff
@@ -1,18 +1,50 @@
 from pydantic import BaseModel, ConfigDict, Field, ValidationError
 
 
+class ApprovedCommandPattern(BaseModel):
+    """Configuration for an approved command pattern.
+
+    Attributes:
+        pattern: The regex pattern to match commands against.
+        description: A human-readable description of what this pattern matches.
+    """
+
+    pattern: str
+    description: str
+
+
 class SecurityConfig(BaseModel):
     """Configuration for security related functionalities.
 
     Attributes:
         confirmation_mode: Whether to enable confirmation mode.
         security_analyzer: The security analyzer to use.
+        approved_command_patterns: List of approved command patterns that don't require confirmation.
+        approved_commands: Dictionary of exact commands that have been approved.
     """
 
     confirmation_mode: bool = Field(default=False)
     security_analyzer: str | None = Field(default=None)
+    approved_command_patterns: list[ApprovedCommandPattern] = Field(
+        default_factory=list
+    )
+    approved_commands: dict[str, bool] = Field(default_factory=dict)
```

Comment:
We don't NEED to save approved commands; we just need to save approved patterns. Also, we also don't need the description field for `ApprovedCommandPattern` -- in this case, can we just greatly simplify ApprovedCommandPattern to be a regex pattern in string format? @openhands